satellite/gc/gc_test.go

```diff
@@ -92,15 +93,30 @@ func TestGarbageCollection(t *testing.T) {
 		time.Sleep(1 * time.Second)
 
 		// Wait for next iteration of garbage collection to finish
+		gcService.Loop.Restart()
 		gcService.Loop.TriggerWait()
 
-		// Check that piece of the deleted object is not on the storagenode
-		pieceAccess, err = targetNode.DB.Pieces().Stat(ctx, storage.BlobRef{
-			Namespace: satellite.ID().Bytes(),
-			Key:       deletedPieceID.Bytes(),
-		})
-		require.Error(t, err)
-		require.Nil(t, pieceAccess)
+		// gc on the storagenode happens in a separate goroutine, so we need to poll until either the piece has been deleted or we time out
+		timeout := 2 * time.Second
+		waitTime := 250 * time.Millisecond
+		startTime := time.Now()
+		for {
+			currTime := time.Now()
+			dt := currTime.Sub(startTime)
+
+			// Check that piece of the deleted object is not on the storagenode
+			pieceAccess, err = targetNode.DB.Pieces().Stat(ctx, storage.BlobRef{
+				Namespace: satellite.ID().Bytes(),
+				Key:       deletedPieceID.Bytes(),
+			})
+			if err != nil && pieceAccess == nil {
```

Comment:
if err != nil is it okay that that err gets reassignd by the Stat call on line 122?

Reply:
yeah, it's fine because this if statement basically takes place of an assertion like `assert.Error(t, err)`. If the error exists, it means garbage collection did what it was supposed to. Do you think there is a way I can make this clearer?